src/terminal/primitives.h

```diff
@@ -893,6 +894,12 @@ constexpr unsigned setDynamicColorCommand(DynamicColorName name)
     }
 }
 
+struct SearchResult
```

Comment:
I think the field members & struct deserve some documentation (even if it might sound trivial to you) :-)

Not sure I get it right, but column is set to the beginning of the match or nullopt if not found. what's remainingText? My point is, maybe `column` should not be optional but both values mandatory and the whole `SearchResult` should be returned by the search functions as `optional<SearchResult>` instead, denoting a "not found" when a `nullopt` was returned. Does sound more intuitive to me.

I'll need to have a second look at this PR later with more time though.